File: src/UserSimulationModel.cpp
#include "UserSimulationModel.hpp"

void UserSimulationModel::initCells(::std::vector<Cell*>& cells)
{
    // TODO: ユーザー独自の処理に変更
    CellSimulationModel::initCells(cells);
}

Vec3 UserSimulationModel::calcCellForce(Cell& c, ::std::vector<Cell*> const& cells, const ::std::vector<UserMoleculeSpace*>& moleculeSpaces)
{
    // TODO: ユーザー独自の処理に変更
    return Vec3::zero();
}

void UserSimulationModel::beforeNextStep(::std::vector<Cell*>& cells, ::std::vector<UserMoleculeSpace*>& moleculeSpaces)
{
    // TODO: ユーザー独自の処理に変更
    CellSimulationModel::beforeNextStep(cells, moleculeSpaces);
}

void UserSimulationModel::onNextStep(::std::vector<Cell*>& cells, ::std::vector<UserMoleculeSpace*>& moleculeSpaces)
{
    // TODO: ユーザー独自の処理に変更
    CellSimulationModel::onNextStep(cells, moleculeSpaces);
}

File: src/cs/CellSim/Users/UserSimulationModel.cs
using CellSim.Cells;
using CellSim.Model;
using CellSim.Numerics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CellSim.Users
{
    /// <summary>
    /// ユーザー定義力学モデル
    /// </summary>
    public class UserSimulationModel : CellSimulationModel
    {
        public UserSimulationModel()
        {
            // TODO: ここに処理を追加します
        }

        // TODO: ComputeForceOnCell()で引数cellAlgorithmを使用する場合は'true'に置き換えます
        public override bool UseCellAlgorithm => false;

        public override void BeforeAdvanceStep(ReadOnlySimulation sender, SimulationModelStepArgs args)
        {
            // TODO: ここに処理を追加します
        }

        public override Vector3 ComputeForceOnCell(ReadOnlySimulation sender, SimulationModelForceComputationArgs args)
        {
            // TODO: ここに処理を追加します

            return Vector3.Zero;
        }

        public override void InitializeCells(ReadOnlySimulation sender, List<Cell> cells)
        {
            // TODO: ここに処理を追加します
            base.InitializeCells(sender, cells);
        }

        public override void OnAdvanceStep(ReadOnlySimulation sender, SimulationModelStepArgs args)
        {
            // TODO: ここに処理を追加します
        }
    }
}
